ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/transaction/UtxoLedgerTransaction.java

```diff
@@ -0,0 +1,288 @@
+     */
+    @NotNull
+    default List<TransactionState<?>> getInputTransactionStates() {
+        return getInputStateAndRefs()
+                .stream()
+                .map(StateAndRef::getState)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Gets the input contract states associated with the current {@link UtxoLedgerTransaction}.
+     *
+     * @return Returns the input contract states associated with the current {@link UtxoLedgerTransaction}.
+     */
+    @NotNull
+    default List<ContractState> getInputContractStates() {
+        return getInputTransactionStates()
+                .stream()
+                .map(TransactionState::getContractState)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Obtains all input contract states that match the specified type.
+     *
+     * @param <T>  The underlying type of the {@link ContractState}.
+     * @param type The type of the {@link ContractState}.
+     * @return Returns all input contract states that match the specified type.
+     */
+    // TODO : Rename to getInputContractStates
+    @NotNull
+    <T extends ContractState> List<T> getInputStates(@NotNull Class<T> type);
+
+    /**
+     * Gets the reference input state refs associated with the current {@link UtxoLedgerTransaction}.
+     *
+     * @return Returns the reference input state refs associated with the current {@link UtxoLedgerTransaction}.
+     */
+    // TODO : Rename to getReferenceInputStateRefs
+    @NotNull
+    List<StateRef> getReferenceStateRefs();
+
+    /**
+     * Gets the reference input states and state refs associated with the current {@link UtxoLedgerTransaction}.
+     *
+     * @return Returns the reference input states and state refs associated with the current {@link UtxoLedgerTransaction}.
+     */
+    // TODO : Rename to getReferenceInputStateAndRefs
+    @NotNull
+    List<StateAndRef<?>> getReferenceStateAndRefs();
+
+    /**
+     * Obtains all reference input states and state refs that match the specified type.
+     *
+     * @param <T>  The underlying type of the {@link ContractState}.
+     * @param type The type of the {@link ContractState}.
+     * @return Returns all reference input states and state refs that match the specified type.
+     */
+    // TODO : Rename to getReferenceInputStateAndRefs
+    @NotNull
+    <T extends ContractState> List<StateAndRef<T>> getReferenceStateAndRefs(@NotNull Class<T> type);
+
+    /**
+     * Gets the reference input transaction states associated with the current {@link UtxoLedgerTransaction}.
+     *
+     * @return Returns the reference input transaction states associated with the current {@link UtxoLedgerTransaction}.
+     */
+    // TODO : Rename to getReferenceInputTransactionStates
+    @NotNull
+    default List<TransactionState<?>> getReferenceTransactionStates() {
+        return getReferenceStateAndRefs()
```

Comment:
Shouldn't affect this PR, but I'd argue we just move most of the default implementations into the actual implementation and streamline the interface code.

Reply:
Na leave it for another time.